### PR TITLE
pythonPackages.toggl-cli: init at 2.1.0 (and its dependencies)

### PR DESCRIPTION
**Pull request by mmahut**
###### Motivation for this change

- pythonPackages.togglCli: init at 2.1.0

And its dependencies:

- pythonPackages.validate_email: init at 1.3
- pythonPackages.readchar: init at 2.0.1
- pythonPackages.inquirer: init at 2.6.3
- pythonPackages.PTable: init at 0.9.2

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by srhb** [comment]
@GrahamcOfBorg build pythonPackages.togglCli

**Comment by srhb** [comment]
@GrahamcOfBorg build python3Packages.togglCli

File: pkgs/development/python-modules/validate_email/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+buildPythonPackage rec {
+  pname = "validate_email";
+  version = "1.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1bxffaf5yz2cph8ki55vdvdypbwkvn2xr1firlcy62vqbzf1jivq";
+  };
+
```
Comment (worldofpeace) [inline pkgs/development/python-modules/validate_email/default.nix]:
```suggestion
  # No Tests
  doCheck = false;
```

File: pkgs/development/python-modules/togglCli/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{ stdenv, buildPythonPackage, fetchPypi, pythonPackages }:
+
+
+buildPythonPackage rec {
+  pname = "togglCli";
+  version = "2.1.0";
+
```
Comment (worldofpeace) [inline pkgs/development/python-modules/togglCli/default.nix]:
```suggestion

  disabled = pythonOlder "3.5";
```

See https://github.com/AuHau/toggl-cli/blob/master/setup.cfg#L10.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Hi @mmahut, what follows is a general review and then a more detailed
one for each commit.

Each package should reference its depedencies directly instead of
using `pythonPackages`. And in general we need to enable tests
which may require fetching from github.

- `pythonPackages.PTable: init at 0.9.2`

Please normalize the attribute name to `ptable` and also `pname`.
###### note you'll need to manually give `fetchPypi` the non normalized `pname`
The commit msg should reflect this change.

Also we should build from the latest commit on GitHub, see https://github.com/kxxoling/PTable/issues/27.
You'll then will be able to enable the [tests](https://github.com/kxxoling/PTable/blob/master/Makefile#L9).


- `pythonPackages.inquirer: init at 2.6.3`

Please enable tests.
Looking at [requirements-dev.txt](https://github.com/magmax/python-inquirer/blob/master/requirements-dev.txt) and the [Makefile](https://github.com/magmax/python-inquirer/blob/master/Makefile) will be helpful.

- `pythonPackages.readchar: init at 2.0.1`

Fetching from PyPi's wheels appear to be to complex and it won't be automatically updated.
Please use `fetchFromGitHub` and enable [tests](https://github.com/magmax/python-readchar/blob/master/setup.py#L67). The [`setup_requires`](https://github.com/magmax/python-readchar/blob/master/setup.py#L78) will need to be in `nativeBuildInputs` btw.

- `pythonPackages.validate_email: init at 1.3`

Please normalize the attribute name to `validate-email` and also `pname`.
The commit msg should reflect this change.

- `pythonPackages.togglCli: init at 2.1.0`

Please normalize the attribute name to `toggle-cli` and also `pname`.
The commit msg should reflect this change.

We also need to correct the dependency declaration for the [`setup_requires`](https://github.com/AuHau/toggl-cli/blob/master/setup.py#L5) `pbr` and `twine`.
They should be `nativeBuildInputs` and `pbr` should also be a `checkInputs` because we need it for tests.

And on the topic of tests we should enable them.

- `Adding pythonPackages.{togglCli,validate_email,readchar,PTable,inquirer}`

Unfortunately this is going to require some git rewriting because each commit that
`init`s should also add itself to `python-packages.nix`.

**Review by worldofpeace — CHANGES_REQUESTED** [review]
I've built everything successfully for all interpreters in nixpkgs :sparkle:

Managed to fix the tests for `toggl-cli` by skipping one's that still somehow needed network access
(probably needs to be fixed upstream)
```diff
diff --git a/pkgs/development/python-modules/toggl-cli/default.nix b/pkgs/development/python-modules/toggl-cli/default.nix
index c44b42b6f2a..5c851fdb475 100644
--- a/pkgs/development/python-modules/toggl-cli/default.nix
+++ b/pkgs/development/python-modules/toggl-cli/default.nix
@@ -24,11 +24,18 @@ buildPythonPackage rec {
   nativeBuildInputs = [ pbr twine ];
   checkInputs = [ pbr pytest pytestcov pytest-mock faker factory_boy ];
 
-  # FIXME: fix tests
-  doCheck = false;
+  preCheck = ''
+     export TOGGL_API_TOKEN=your_api_token
+     export TOGGL_PASSWORD=toggl_password
+     export TOGGL_USERNAME=user@example.com
+  '';
 
   checkPhase = ''
-   pytest -m "not premium" tests/unit
+    runHook preCheck
+
+    pytest -k "not premium and not TestDateTimeType and not TestDateTimeField" tests/unit --maxfail=20
+
+    runHook postCheck
   '';
 
   propagatedBuildInputs = [

```
Also pr'd something for the readchar pin in inquirer https://github.com/magmax/python-inquirer/pull/77

Following comments are just some expression cleanup.

**Comment by mmahut** [comment]
@worldofpeace you are amazing - thank you for your review! Pushed the new changes.

**Comment by worldofpeace** [comment]
Great work here @mmahut :sparkles: 

I think this is complete, can you squash the commits?

**Comment by mmahut** [comment]
@worldofpeace thank you! Done!

**Pull request merged**